### PR TITLE
Insert Entrypoint hook invocation logic

### PR DESCRIPTION
**Pull request by schernysh**

### CODE REVIEW

File: src/main/java/org/prebid/server/exception/RejectedRequestException.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.prebid.server.exception;
+
+import org.prebid.server.hooks.execution.model.HookExecutionContext;
+
+public class RejectedRequestException extends RuntimeException {
+
+    private final HookExecutionContext hookExecutionContext;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/exception/RejectedRequestException.java:7]: Why we need it here?

File: src/main/java/org/prebid/server/hooks/execution/model/GroupExecutionOutcome.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.prebid.server.hooks.execution.model;
+
+import lombok.Value;
+
+import java.util.List;
+
+@Value(staticConstructor = "of")
+public class GroupExecutionOutcome {
+
+    List<HookExecutionOutcome> hooks;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/hooks/execution/model/GroupExecutionOutcome.java:10]: Should it be like `hookOutcomes`?
Also, if yes - pls check StageExecutionOutcome.groups.

File: src/main/java/org/prebid/server/hooks/execution/model/Stage.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.prebid.server.hooks.execution.model;
+
+public enum Stage {
+
+    entrypoint, auction_request, bidder_request, bidder_response, auction_response
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/hooks/execution/model/Stage.java:5]: Pls align with updated stages:
- rename auction_request to raw_auction_request
- add processed_auction_request
- rename bidder_response to raw_bidder_response
- add processed_bidder_response

**Review by rpanchyk — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/exception/RejectedRequestException.java (hunk 1 above)
Comment (schernysh) [reply src/main/java/org/prebid/server/exception/RejectedRequestException.java:7]: In order to get debug information from context and put it in response.

File: src/main/java/org/prebid/server/hooks/execution/model/GroupExecutionOutcome.java (hunk 2 above)
Comment (schernysh) [reply src/main/java/org/prebid/server/hooks/execution/model/GroupExecutionOutcome.java:10]: I picked short name on purpose as `executionOutcome` or `outcome` suffix doesn't add any useful bits. It's already in Stage or GroupExecutionContext so there is no reason to repeat the suffix in my opinion. Next PR's exploring usage of these models may prove this assumption wrong of course so it makes sense to defer final decision for later.

**Review by rpanchyk — APPROVED** [review]

**Review by DGarbar — APPROVED** [review]

**Pull request merged**